pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,9 +120,58 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .map(m -> {
+                               Integer a = xpathExpressionArea.getText().lastIndexOf("/", m.getInsertionEnd());
```

Comment:
This line gets the index of the last slash after the insertion point. It should get the index of the first slash before the insertion end *looking right-to-left*! Standard String methods can't help here and you should implement your own method that loops from right to left until you find the character you're looking for, starting from a particular index.

For God's sake use better names for your variables. If you did so you'd probably understand your own code better. 

And btw you shouldn't use `Integer` but `int`

Reply:
Sorry , I'll use better names